idaes/models/unit_models/packed_bed_reactor.py

```diff
@@ -0,0 +1,337 @@
+        doc="""bulk density of the catalyst,
+**default** - 1.1"""))
+    def build(self):
+        """
+        Begin building model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(PBRData, self).build()
+
+        # Build Control Volume
+        self.control_volume = ControlVolume1DBlock(default={
+                "dynamic": self.config.dynamic,
+                "has_holdup": self.config.has_holdup,
+                "property_package": self.config.property_package,
+                "property_package_args": self.config.property_package_args,
+                "reaction_package": self.config.reaction_package,
+                "reaction_package_args": self.config.reaction_package_args,
+                "transformation_method": self.config.transformation_method,
+                "transformation_scheme": self.config.transformation_scheme,
+                "finite_elements": self.config.finite_elements,
+                "collocation_points": self.config.collocation_points})
+
+        self.control_volume.add_geometry(
+                length_domain_set=self.config.length_domain_set)
+
+        self.control_volume.add_state_blocks(
+                has_phase_equilibrium=self.config.has_phase_equilibrium)
+
+        self.control_volume.add_reaction_blocks(
+                has_equilibrium=self.config.has_equilibrium_reactions)
+
+        self.control_volume.add_material_balances(
+            balance_type=self.config.material_balance_type,
+            has_rate_reactions=True,
+            has_equilibrium_reactions=self.config.has_equilibrium_reactions,
+            has_phase_equilibrium=self.config.has_phase_equilibrium)
+
+        self.control_volume.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_of_reaction=self.config.has_heat_of_reaction,
+            has_heat_transfer=self.config.has_heat_transfer)
+
+        self.control_volume.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=self.config.has_pressure_change)
+
+        self.control_volume.apply_transformation()
+
+        # Add Ports
+        self.add_inlet_port()
+        self.add_outlet_port()
+
+        # Set references to balance terms at unit level
+        add_object_reference(self, "length", self.control_volume.length)
+
+        # Add volume variable for full reactor
+        units = self.config.property_package.get_metadata()
+        self.volume = Var(initialize=1,
+                          doc="Reactor Volume",
+                          units=units.get_derived_units("volume"))
+
+        self.area = Var(initialize=1,
+                        doc="area",
+                        units=units.get_derived_units("area"))
+
+        self.geometry = Constraint(expr=self.volume == self.config.mass_catalyst/self.config.bulk_density)
```

Comment:
You should never use entries from the config block in an expression - if you do this it means that they can never be changed from the initial value in the config block as they will be interpreted a constants.

As mentioned earlier, what you should do is:

1. Remove these entries from the config block.
2. Create new `Var` components for these on the model, along with appropriate units (e.g. for density)
3. Use these new components when constructing the required expressions.
4. Users will then need to fix the values for these `Vars` as inputs when solving their model, which also allows them to be adjusted if needed or (even optimized in future if appropriate).